test/boost/sstable_3_x_test.cc

```diff
@@ -2860,6 +2884,60 @@ SEASTAR_THREAD_TEST_CASE(test_uncompressed_simple_read) {
                       {{int_cdef, int32_type->decompose(1003)}})
         .produces_partition_end()
         .produces_end_of_stream();
+
+    // Reversed reader should give exactly the same result as each partition has
+    // only one row
+
+    {
+    auto range = dht::partition_range::make_singular(to_key(5));
+    assert_that(sst.make_reversing_reader(range))
+        .produces_partition_start(to_key(5))
+        .produces_row(clustering_key::from_single_value(*UNCOMPRESSED_SIMPLE_SCHEMA,
+                                                        int32_type->decompose(105)),
+                      {{int_cdef, int32_type->decompose(1005)}})
+        .produces_partition_end()
+        .produces_end_of_stream();
+    }
+    {
+    auto range = dht::partition_range::make_singular(to_key(1));
+    assert_that(sst.make_reversing_reader(range))
+        .produces_partition_start(to_key(1))
+        .produces_row(clustering_key::from_single_value(*UNCOMPRESSED_SIMPLE_SCHEMA,
+                                                        int32_type->decompose(101)),
+                      {{int_cdef, int32_type->decompose(1001)}})
+        .produces_partition_end()
+        .produces_end_of_stream();
+    }
+    {
+    auto range = dht::partition_range::make_singular(to_key(2));
+    assert_that(sst.make_reversing_reader(range))
+        .produces_partition_start(to_key(2))
+        .produces_row(clustering_key::from_single_value(*UNCOMPRESSED_SIMPLE_SCHEMA,
+                                                        int32_type->decompose(102)),
+                      {{int_cdef, int32_type->decompose(1002)}})
+        .produces_partition_end()
+        .produces_end_of_stream();
+    }
+    {
+    auto range = dht::partition_range::make_singular(to_key(4));
+    assert_that(sst.make_reversing_reader(range))
+        .produces_partition_start(to_key(4))
+        .produces_row(clustering_key::from_single_value(*UNCOMPRESSED_SIMPLE_SCHEMA,
+                                                        int32_type->decompose(104)),
+                      {{int_cdef, int32_type->decompose(1004)}})
+        .produces_partition_end()
+        .produces_end_of_stream();
+    }
+    {
+    auto range = dht::partition_range::make_singular(to_key(3));
+    assert_that(sst.make_reversing_reader(range))
+        .produces_partition_start(to_key(3))
+        .produces_row(clustering_key::from_single_value(*UNCOMPRESSED_SIMPLE_SCHEMA,
+                                                        int32_type->decompose(103)),
```

Comment:
Need to test more complicated schemas. Are the partitions large enough that the promoted index is exercised?

Reply:
(need to test both cases, and need to test a read starting at every row of a large partition so we find edge cases)

Reply:
> Are the partitions large enough that the promoted index is exercised?

Probably not in this test, but it is in `test_sstable_reversing_conforms_to_mutation_source` (in `test/boost/sstable_conforms_to_mutation_source_test.cc`) as the test there is executed for different values of `sstable_writer_config::promoted_index_block_size`. It actually already managed to catch a bug which manifested for block size = 128 but not for block size =  1.